eng/generators.targets

```diff
@@ -57,9 +57,7 @@
                           '$(MSBuildProjectName)' == 'System.Private.CoreLib'
                           or '$(EnableDllImportGenerator)' == 'true'
                           or ('@(Reference)' != ''
-                            and (@(Reference->AnyHaveMetadataValue('Identity', 'System.Runtime.CompilerServices.Unsafe'))
-                              or ('@(ProjectReference)' != ''
-                                and @(ProjectReference->AnyHaveMetadataValue('Identity', $([MSBuild]::NormalizePath('$(LibrariesProjectRoot)', 'System.Runtime.CompilerServices.Unsafe', 'src', 'System.Runtime.CompilerServices.Unsafe.ilproj'))))))
+                            and (@(Reference->AnyHaveMetadataValue('Identity', 'System.Runtime.CompilerServices.Unsafe')))
```

Comment:
If you remove the requirement to reference System.Runtime.CompilerServices.Unsafe here, you'll fix the problem.

The issue is that you're still requiring asemblies to reference System.Runtime.CompilerServices.Unsafe to include the `ArrayMarshaller`, but you've updated the projects to not reference it as the types now live in `System.Runtime`. If you remove this part of the condition, things will work as expected.

```suggestion
```

Reply:
Thanks. That got me to the next set of weirdness which I've traced and I think resolved.